### PR TITLE
feat: add opm support for optional validators and custom category validation

### PR DESCRIPTION
**Pull request by exdx**
**Description of the change:**
Bumps the `api` dependency to incorporate the latest optional validators: the bundle object validator and the new operatorhub validator that enables usage of a custom categories file. 

Documented at https://olm.operatorframework.io/docs/tasks/validate-package/#validation (will update once this PR goes in and the flags are definite) 

**Motivation for the change:**

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: cmd/opm/alpha/bundle/validate.go [hunk 1]
```diff
@@ -34,6 +47,8 @@ accurate.`,
 	}
 
 	bundleValidateCmd.Flags().StringVarP(&containerTool, "image-builder", "b", "docker", "Tool used to pull and unpack bundle images. One of: [none, docker, podman]")
+	bundleValidateCmd.Flags().StringVarP(&optional, "optional-validators", "o", "", "Specifies optional validations (operatorhub validation, bundle-objects validation, etc) to be run. For example, set to operatorhub,bundle-objects "+
```
Comment (kevinrizza) [inline cmd/opm/alpha/bundle/validate.go:50]: Can we add an enumeration to the docs text for the set of valid values here?

**Review by kevinrizza — CHANGES_REQUESTED** [review]

**Comment by exdx** [comment]
This is looking pretty much ready, but requires that the api view of the bundle and the registry view of the bundle be the same (so the registry can pass the api bundle down to the operatorhub validator). Will refactor in a separate PR so that there is only one consistent bundle definition living in api. 

/hold

**Comment by exdx** [comment]
Decided to simply go with the manifest bundle type for the operatorhub validator (its a very straightforward validation - just looks at the versioned CSV object) and hold off on the refactor for now. The validators work as intended now.

```
operator-registry(feat/custom-validation-categories) ✗: OPERATOR_BUNDLE_CATEGORIES=../api/pkg/validation/internal/testdata/categories.json ./bin/opm alpha bundle validate --tag quay.io/dsover/bundle-categories:latest --image-builder docker -o operatorhub,bundle-object
INFO[0000] Create a temp directory at /var/folders/9z/c4spd5wn0q51kw278nmpsjpm0000gn/T/bundle-913462645  container-tool=docker
DEBU[0000] Pulling and unpacking container image         container-tool=docker
INFO[0000] running /usr/local/bin/docker pull quay.io/dsover/bundle-categories:latest  container-tool=docker
INFO[0001] running docker create                         container-tool=docker
DEBU[0001] [docker create quay.io/dsover/bundle-categories:latest ]  container-tool=docker
INFO[0001] running docker cp                             container-tool=docker
DEBU[0001] [docker cp c34e972aa2841489317a82b403252bf48d9795f3c194cc76891c20bbd0aa68cc:/. /var/folders/9z/c4spd5wn0q51kw278nmpsjpm0000gn/T/bundle-913462645]  container-tool=docker
INFO[0001] running docker rm                             container-tool=docker
DEBU[0001] [docker rm c34e972aa2841489317a82b403252bf48d9795f3c194cc76891c20bbd0aa68cc]  container-tool=docker
INFO[0001] Unpacked image layers, validating bundle image format & contents  container-tool=docker
DEBU[0001] Found manifests directory                     container-tool=docker
DEBU[0001] Found metadata directory                      container-tool=docker
DEBU[0001] Getting mediaType info from manifests directory  container-tool=docker
INFO[0001] Found annotations file                        container-tool=docker
INFO[0001] Could not find optional dependencies file     container-tool=docker
DEBU[0001] Validating bundle contents                    container-tool=docker
DEBU[0001] Validating "apiextensions.k8s.io/v1beta1, Kind=CustomResourceDefinition" from file "etcdbackups.etcd.database.coreos.com.crd.yaml"  container-tool=docker
DEBU[0001] Validating "apiextensions.k8s.io/v1beta1, Kind=CustomResourceDefinition" from file "etcdclusters.etcd.database.coreos.com.crd.yaml"  container-tool=docker
DEBU[0001] Validating "operators.coreos.com/v1alpha1, Kind=ClusterServiceVersion" from file "etcdoperator.v0.9.4.clusterserviceversion.yaml"  container-tool=docker
DEBU[0001] Validating "apiextensions.k8s.io/v1beta1, Kind=CustomResourceDefinition" from file "etcdrestores.etcd.database.coreos.com.crd.yaml"  container-tool=docker
DEBU[0001] Performing operatorhub validation             container-tool=docker
INFO[0001] All validation tests have been completed successfully  container-tool=docker
```

/hold cancel

**Review by ankitathomas — COMMENTED** [review]
/lgtm

**Comment by exdx** [comment]
bumping api dep for category bugfix
/hold

**Comment by exdx** [comment]
/hold cancel 
rebased off v0.3.22

**Review by hasbro17 — APPROVED** [review]
/lgtm

**Pull request merged**